app/models/service_template_ansible_playbook.rb

```diff
@@ -163,4 +165,24 @@ def destroy
     end
     super
   end
+
+  # ServiceTemplate includes a retirement resource action
+  #   with a defined job template:
+  #
+  #   1. A resource_action that includes a configuration_template_id.
+  #   2. At least one service instance where :retired is set to false.
+  #
+  def potential_retirement?
+    retirement_jt_exists = resource_actions.where("action = 'Retirement' AND configuration_template_id IS NOT NULL").present?
+    retirement_jt_exists && services.where(:retired => false).exists?
+  end
+
+  private
+
+  def check_for_retirement_potential
```

Comment:
Two minor suggestions:

1. The word "_for_" is not needed.  `check_retirement_potential`.
2. Rename the other method `retirement_potential?' instead of swapping the order.